codemagic.yaml

```diff
@@ -0,0 +1,127 @@
+workflows:
+    deploy_staging:
+      name: Deploy Firebase Staging
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'develop'
+            include: true
+        cancel_previous_builds: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match Ad-hoc
+          script: bundle exec fastlane syncAdHocStagingCodeSigning
+        - name: Build App and Distribute to Firebase
+          script: bundle exec fastlane buildStagingAndUploadToFirebase
+      artifacts:
+        - ./*.ipa
+        - ./*.dSYM.zip
+    deploy_release_firebase:
+      name: Deploy Firebase Production
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'release/*'
+            include: true
+        cancel_previous_builds: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match Ad-hoc
+          script: bundle exec fastlane syncAdHocProductionCodeSigning
+        - name: Build App and Distribute to Firebase
+          script: bundle exec fastlane buildProductionAndUploadToFirebase
+      artifacts:
+        - ./*.ipa
+        - ./*.dSYM.zip
+    deploy_app_store:
+      name: Deploy App Store
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'main'
+            include: true
+        cancel_previous_builds: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match Ad-hoc
+          script: bundle exec fastlane syncAdHocProductionCodeSigning
```

Comment:
I think we should remove this step and add a note to this PR 🤔

Reply:
Note updated 👍